Crash after removing image while viewing image

I finally found it! By pure chance! The bug I've been seeing in Google Play Console for years that *nobody* ever reported! It finally makes sense!

1. Ensure the card has a front image
2. Open the card
3. Press right to switch the main image display to front image
4. Edit card
5. Remove front image
6. Save
7. Catima crashes

```
FATAL EXCEPTION: main
Process: me.hackerchick.catima, PID: 2132
java.lang.RuntimeException: Unable to resume activity {me.hackerchick.catima/protect.card_locker.LoyaltyCardViewActivity}: java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 1
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5244)
	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:5277)
	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:57)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:60)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleItem(TransactionExecutor.java:282)
	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:150)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:93)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2595)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8592)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:878)
Caused by: java.lang.IndexOutOfBoundsException: Index 1 out of bounds for length 1
	at jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64)
	at jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70)
	at jdk.internal.util.Preconditions.checkIndex(Preconditions.java:266)
	at java.util.Objects.checkIndex(Objects.java:359)
	at java.util.ArrayList.get(ArrayList.java:434)
	at protect.card_locker.LoyaltyCardViewActivity.drawMainImage(LoyaltyCardViewActivity.java:968)
	at protect.card_locker.LoyaltyCardViewActivity.setFullscreen(LoyaltyCardViewActivity.java:1169)
	at protect.card_locker.LoyaltyCardViewActivity.onResume(LoyaltyCardViewActivity.java:763)
	at android.app.Instrumentation.callActivityOnResume(Instrumentation.java:1722)
	at android.app.Activity.performResume(Activity.java:9118)
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:5234)
	... 14 more
```

--- a/app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java
+++ b/app/src/main/java/protect/card_locker/LoyaltyCardViewActivity.java
@@ -1085,6 +1085,12 @@ public class LoyaltyCardViewActivity extends CatimaAppCompatActivity implements 
     }
 
     private void setMainImagePreviousNextButtons() {
+        // Ensure the main image index is valid. After a card update, some images (front/back/barcode)
+        // may have been removed, so the index should not exceed the number of available images.
+        if(mainImageIndex > imageTypes.size() - 1){
+            mainImageIndex = 0;
+        }
+
         if (imageTypes.size() < 2) {
             binding.mainLeftButton.setVisibility(View.INVISIBLE);
             binding.mainRightButton.setVisibility(View.INVISIBLE);

